pulp_container/app/modelresource.py

```diff
@@ -1,7 +1,49 @@
 from import_export import fields, widgets
 from pulpcore.plugin.importexport import QueryModelResource, BaseContentResource
+from pulpcore.plugin.modelresources import RepositoryResource
 
-from pulp_container.app.models import Blob, Manifest, ManifestListManifest, ManifestSignature, Tag
+from pulp_container.app.models import (
+    Blob,
+    ContainerRepository,
+    ContainerPushRepository,
+    Manifest,
+    ManifestListManifest,
+    ManifestSignature,
+    Tag,
+)
+
+
+class ContainerRepositoryResource(RepositoryResource):
+    """
+    A resource for importing/exporting repositories of the sync type
+    """
+
+    def set_up_queryset(self):
+        """
+        :return: A queryset containing one sync repository that will be exported.
+        """
+        return ContainerRepository.objects.filter(pk=self.repo_version.repository)
+
+    class Meta:
+        model = ContainerRepository
+        exclude = RepositoryResource.Meta.exclude + ("manifest_signing_service",)
+
+
+class ContainerPushRepositoryResource(RepositoryResource):
+    """
+    A resource for importing/exporting repositories of the push type
+    """
+
+    def set_up_queryset(self):
+        """
+        :return: A queryset containing one push repository that will be exported.
+        """
+        return ContainerPushRepository.objects.filter(pk=self.repo_version.repository)
+
+    class Meta:
+        # import the repository as a repository of the sync type
+        model = ContainerRepository
```

Comment:
This needs to be reassessed because there might be a conflict between the types that import-export framework is trying to create. In particular, the framework tries to create a sync-repository even though an initial push-repository could still exist during the import procedure. As a result, the import fails on a duplicated key error (e.g., repository "foo" already exists)...

Reply:
And i think it's safe to fail in such circumstances. Just assuming the user may have wanted something they didn't ask for is way to dangerous.